core/shared/platform/include/platform_api_extension.h

```diff
@@ -977,6 +977,19 @@ os_socket_set_broadcast(bh_socket_t socket, bool is_enabled);
 int
 os_socket_get_broadcast(bh_socket_t socket, bool *is_enabled);
 
+/**
+ * Dump memory information of the current process
+ * It may have variant implementations in different platforms
+ *
+ * @param out the output buffer. It is for sure the return content
+ *            is a c-string which ends up with '\0'
+ * @param size the size of the output buffer
+ *
+ * @return 0 if success, -1 otherwise
+ */
+int
+os_dumps_proc_mem_info(char *out, unsigned int size);
```

Comment:
os_dump_proc_mem_info

Reply:
like `fgets` means get as a string?

Reply:
Other places we use bh_print_proc_mem, bh_log_proc_mem, wasm_runtime_dump_module_mem_consumption, wasm_runtime_dump_perf_profiling, using `dumps` makes inconsistent :)